experimental/framework/data-objects/api-report/data-objects.api.md

```diff
@@ -22,31 +23,18 @@ export interface IRuntimeSignaler {
 }
 
 // @internal
-export interface ISignaler {
+export interface ISignaler extends IEventProvider<IErrorEvent> {
     offSignal<T>(signalName: string, listener: SignalListener<T>): ISignaler;
     onSignal<T>(signalName: string, listener: SignalListener<T>): ISignaler;
     submitSignal<T>(signalName: string, payload?: Jsonable<T>): any;
 }
 
 // @internal
-export class Signaler extends DataObject<{
-    Events: IErrorEvent;
-}> implements EventEmitter, ISignaler {
-    // (undocumented)
-    static readonly factory: DataObjectFactory<Signaler, {
-        Events: IErrorEvent;
-    }>;
-    // (undocumented)
-    protected hasInitialized(): Promise<void>;
-    // (undocumented)
-    static readonly Name = "@fluid-example/signaler";
-    // (undocumented)
-    offSignal<T>(signalName: string, listener: SignalListener<T>): ISignaler;
-    // (undocumented)
-    onSignal<T>(signalName: string, listener: SignalListener<T>): ISignaler;
-    // (undocumented)
-    submitSignal<T>(signalName: string, payload?: Jsonable<T>): void;
-}
+export const Signaler: {
+    readonly factory: IFluidDataStoreFactory & {
+        readonly registryEntry: NamedFluidDataStoreRegistryEntry;
+    };
+} & SharedObjectKind<ISignaler>;
```

Comment:
Signaler is not a "SharedObject" - what does it mean for it to be SharedObjectKind semantically?  Is this pointing to a need for a rename of SharedObjectKind, or something else?

Reply:
SharedObjectKind is the abstraction used for things that can go in ContainerSchema. While support for DataObjects there is working (and thats what this is using) we don't explicitly refer to "DataObject" in the names as no data object specific stuff is public in the declarative API.

If you have a better name than SharedObjectKind for this concept in the declarative API surface, that could be a useful rename (in a different change). I don't think such a rename is too important from the public API perspective though as there isn't really any difference for how data objects vs shared objects are used since we don't publicly expose SharedObject or DataObject as concepts at that layer, only this single unified type.

Reply:
To be clear, I do think a better name would be nice, as this name might be fine from the public API, it creates some confusion internally. I named it and I have no objection to it going through a rename, though I personally don't want to take on the work of that process myself at the moment.